preinstall_check.js

```diff
@@ -0,0 +1,32 @@
+const isUsingNpm = process.env.npm_config_git !== undefined;
```

Comment:
When I searched around, this seemed like the simplest way to know it was `npm` and not `yarn`. E.g. from yarn team lead: https://twitter.com/cpojer/statuses/953286334013231104